### PR TITLE
Add smart case function

### PR DESCRIPTION
**Pull request by cychen2021**
When the option EASY_MOTION_SMART_CASE on, the plugin will match both the original lower case letter and the corresponding upper case letter if a lower case letter is provided.

It is just like the smart case option for the original vim easymotion plugin.

### CODE REVIEW

File: easy_motion.py [hunk 1]
```diff
@@ -401,7 +409,10 @@ def reset_terminal(old_term_settings):
             elif read_state == ReadState.HIGHLIGHT:
                 assert motion is not None
                 if grouped_indices is None:
-                    indices = motion_to_indices(cursor_position, text, motion, motion_argument)
+                    if not PY2 and smart_case and motion_argument is not None and motion_argument.islower():
```
Comment (IngoMeyer441) [inline easy_motion.py]: Why is Python 2 excluded? `chain` is also [available in Python 2](https://docs.python.org/2.7/library/itertools.html#itertools.chain).

File: easy_motion.py [hunk 2]
```diff
@@ -401,7 +409,10 @@ def reset_terminal(old_term_settings):
...
+                      indices = chain(motion_to_indices(cursor_position, text, motion, motion_argument), motion_to_indices(cursor_position, text, motion, motion_argument.upper()))
```
Comment (IngoMeyer441) [inline easy_motion.py]: It is correct to put both results together, but the assigned target keys won't be in the correct order. `motion_to_indices` sorts the results by the distance from the current cursor position. IMHO, the results of both calls should be merged together to preserve the order or the regular expressions in `motion_to_indices` should be used with the case-insensitive option, which is probably easier.

**Review by IngoMeyer441 — CHANGES_REQUESTED** [review]
Thanks for your PR! I have some minor issues. Would be great if you could look into this before the PR is merged.

File: easy_motion.py (hunk 1 above)
Comment (cychen2021) [reply easy_motion.py]: It's a mistake. I'm unfamiliar with Python 2 and thought it didn't supported this. I'll fix it.

File: easy_motion.py (hunk 2 above)
Comment (cychen2021) [reply easy_motion.py]: I'll fix this in later commits.

**Comment by cychen2021** [comment]
Ready for review @IngoMeyer441

**Comment by IngoMeyer441** [comment]
Thank you very much for your contribution, looks good to me!

**Pull request merged**